opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -175,7 +176,7 @@ namespace Opm {
         /// \param[in, out] reservoir_state   reservoir state variables
         /// \param[in, out] well_state        well state variables
         template <class NonlinearSolverType>
-        IterationReport nonlinearIteration(const int iteration,
+        SimulatorReport nonlinearIteration(const int iteration,
```

Comment:
simple: on one side, there is the report which is returned by the `step()` method, on the other side, there are `numIterations()` and friends which are completely orthogonal and must be used in case of exceptions...